contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -743,6 +743,13 @@ Incorporating the default values into the `Spec` ensures that `Spec` depicts the
 full desired state so that it is easier for the system to determine how to
 achieve the state, and for the user to know what to anticipate.
 
+Default values can be specified on a field using the `+default=` tag. Primitives
+will have their values directly assigned while structs will go through the
+JSON unmarshalling process. Fields that do not have an `omitempty` json tag may
+only default to the zero value of their corresponding type.
```

Comment:
```suggestion
JSON unmarshalling process. Fields that do not have an `omitempty` json tag will
default to the zero value of their corresponding type if no default is assigned.
```

Reply:
On a second pass, this doesn't seem to be correct. I'd like to say that fields that do not have an omitempty tag will default to the zero value regardless of whether a default is set or not. If the default is set, it must be set to the zero value or else an error will occur.

Reply:
Why would we make it work that way? "omitempty" is meaningless for fields with a default.

Reply:
Oops sorry I missed this earlier. My explanation was a bit unclear. 

Yes, omitempty is meaningless when default is set, but there is a restriction of what the default value can be.

With the omitempty tag, the default value can be any value (within the domain of the type). Without the omitempty tag, the default value MUST be set to the to the zero value. This is because go structs for fields that are not pointers will automatically set the zero value for the field.

I still think the original explanation is the most accurate, although maybe a bit convoluted.